optihood/buildings.py

```diff
@@ -1,392 +1,383 @@
+
+            elif opt == "env":
+                epc=s["impact_cap"] / s["lifetime"]
+                base=0
+                env_capa= s["impact_cap"] / s["lifetime"]
+                env_flow=s["heat_impact"]
+                varc= s["heat_impact"] # variable cost is only passed for environmental optimization if there are emissions per kWh of energy produced from the unit
+
+                envParam = [s["heat_impact"], 0, env_capa]
+
+            collector=SolarCollector(s["label"], self.__buildingLabel,
+                                                   self.__busDict[s["from"] + '__' + self.__buildingLabel],
+                                                   self.__busDict[s["to"] + '__' + self.__buildingLabel],
+                                                   self.__busDict[s["connect"]+ '__' + self.__buildingLabel],
+                                                   s["electrical_consumption"], s["peripheral_losses"], s["latitude"],
+                                                   s["longitude"], s["tilt"], s["roof_area"],
+                                                   s["zenith_angle"], s["azimuth"],
+                                                   s["eta_0"], s["a_1"], s["a_2"], s["temp_collector_inlet"],
+                                                   s["delta_temp_n"], data_timeseries['gls'], data_timeseries['str.diffus'],
+                                                    data_timeseries['tre200h0'], s["capacity_min"], s["capacity_max"],
+                                                   epc, base, env_capa, env_flow, varc)
+            self.__nodesList.append(collector.getSolar("source"))
+            self.__nodesList.append(collector.getSolar("transformer"))
+            self.__nodesList.append(collector.getSolar("sink"))
+
+            self.__envParam['solarConnectBus' + '__' + self.__buildingLabel] = envParam
+
+            self.__costParam['solarConnectBus' + '__' + self.__buildingLabel] = [self._calculateInvest(s)[0],
+                                                                          self._calculateInvest(s)[1]]
+            self.__technologies.append(
+                [s["to"] + '__' + self.__buildingLabel, s["label"] + '__' + self.__buildingLabel])
+
+    def addGridSeparation(self, dataGridSeparation):
+        if not dataGridSeparation.empty:
+            for i, gs in dataGridSeparation.iterrows():
+                self.__nodesList.append(solph.Transformer(label=gs["label"]+'__'+self.__buildingLabel,
+                                                          inputs={self.__busDict[gs["from"]+'__'+self.__buildingLabel]: solph.Flow()},
+                                                          outputs={self.__busDict[gs["to"]+'__'+self.__buildingLabel]: solph.Flow()},
+                                                  conversion_factors={self.__busDict[gs["to"]+'__'+self.__buildingLabel]: gs["efficiency"]}))
+
+    def addSource(self, data, data_elec, data_cost, opt):

```

Comment:
data_cost and data_elec both should be optional parameters. If there is a time series for electricity cost and impact, then this time series should be used, otherwise static cost and impact values should be used. The absence of time series should not raise an error.

Reply:
@vjacquot : this is also a priority. we decided to change the location of the file path for electricity and gas cost files in 'commodity sources' in place of the constant cost. Could you also please have an printed error message if the cell has a incorrect value (non numeric or non path or no file) ?